pkg/collector/osmlogs_collector.go

```diff
@@ -34,34 +34,29 @@ func (collector *OsmLogsCollector) Collect() error {
 	// can define resources to query in deployment.yaml and iterate through the commands+resources needed and create multiple files
 	// see kubeobject collector for example
 
-	// * Get osm resources as table
-	allResourcesFile := filepath.Join(rootPath, "allResources")
-	output, err := utils.RunCommandOnContainer("kubectl", "get", "all", "--all-namespaces", "--selector", "app.kubernetes.io/name=openservicemesh.io")
+	// ** Get ground truth on OSM resources **
+	// * Get osm resources as table *
+	collectResourceToFile(collector, rootPath, "allResourcesTable", "all", "--all-namespaces", "--selector=app.kubernetes.io/name=openservicemesh.io", "-o=wide")
 	if err != nil {
 		return err
 	}
 
-	err = utils.WriteToFile(allResourcesFile, output)
+	// * Get osm resources configs *
+	collectResourceToFile(collector, rootPath, "allResourcesConfigs", "all", "--all-namespaces", "--selector=app.kubernetes.io/name=openservicemesh.io", "-o=json")
 	if err != nil {
 		return err
 	}
 
-	collector.AddToCollectorFiles(allResourcesFile)
-
-	// * Get osm resource configs
-	allResourceConfigsFile := filepath.Join(rootPath, "allResourceConfigs")
-	output, err = utils.RunCommandOnContainer("kubectl", "get", "all", "--all-namespaces", "--selector", "app.kubernetes.io/name=openservicemesh.io", "-o", "json")
+	// * Get webhooks *
+	collectResourceToFile(collector, rootPath, "mutatingWebhookConfigurations", "MutatingWebhookConfiguration", "--all-namespaces", "--selector=app.kubernetes.io/name=openservicemesh.io", "-o=json")
 	if err != nil {
 		return err
 	}
-
-	err = utils.WriteToFile(allResourceConfigsFile, output)
+	collectResourceToFile(collector, rootPath, "validatingWebhookConfigurations", "ValidatingWebhookConfiguration", "--all-namespaces", "--selector=app.kubernetes.io/name=openservicemesh.io", "-o=json")
```

Comment:
Ideally it would be great to use a [variadic function](https://medium.com/rungo/variadic-function-in-go-5d9b23f4c01a) to pass in a variable number of string as args here. The is useful eg: when a selector is not needed in the collectNamespacesMetadata function. But when I tried using this, it converted to a slice. I also tried passing in a slice and converting it to a variadic but this is mysteriously not working without giving me an error message. This would be a very helpful change to clean up all our funcs